src/qt/transactiontablemodel.cpp

```diff
@@ -82,6 +83,7 @@ class TransactionTablePriv
                 }
             }
         }
+        parent->endResetModel();
```

Comment:
would it be a valid state of parent if an exception would be thrown before `endResetModel()` is called?

Reply:
Hmm.. Not sure I get the question.. What kind of exception? And why you think it should affect `parent` (`TransactionTableModel*`)?

Reply:
https://doc.qt.io/qt-6/qabstractitemmodel.html#endResetModel
> You must call this function after resetting any internal data structure in your model or proxy model.


I see that some methods doesn't have `noexcept` specifier, more over functions between beginResetModel() and endResetModel() such as `getWalletTxs`, `showTransaction` have allocation inside: it means that any time an exception can be thrown.

There're two possible solution for resolve "you must call" request:
 - write a wrapper such as `lock_guard` that will call endResetModel() in a destructor.
 - wrap everything between begin & endResetModel to try-catch and safely call `endResetModel`